compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/AnnotationReference.kt

```diff
@@ -55,23 +55,24 @@ public sealed class AnnotationReference {
     public val annotation: KtAnnotationEntry,
     override val classReference: ClassReference
   ) : AnnotationReference() {
-    override fun boundTypeOrNull(): FqName? {
+    override fun boundTypeOrNull(): ClassReference? {
       return annotation.findAnnotationArgument<KtClassLiteralExpression>(
         name = "boundType",
         index = 1
-      )?.requireFqName(module)
+      )?.requireFqName(module)?.toClassReference(module)
     }
   }
 
   public class Descriptor internal constructor(
     public val annotation: AnnotationDescriptor,
     override val classReference: ClassReference
   ) : AnnotationReference() {
-    override fun boundTypeOrNull(): FqName? {
+    override fun boundTypeOrNull(): ClassReference? {
       return (annotation.getAnnotationValue("boundType") as? KClassValue)
         ?.argumentType(module)
         ?.classDescriptor()
         ?.fqNameSafe
+        ?.toClassReference(module)
```

Comment:
That's cheaper to go from the descriptor to the class reference.
```suggestion
        ?.classDescriptor()
        ?.toClassReference(module)
```

Reply:
Just curious, why is this cheaper? Does FqName.toClassReference need to figure out psi vs descriptor at some point?

Reply:
If we only have an `FqName`, then we try to resolve the class descriptor first. We need to know that the `FqName` can be mapped to a class . This is the expensive operation. Since you already have a descriptor here, you can skip this step.